src/systemd/cockpit-wsinstance-http.service.in

```diff
@@ -9,3 +9,17 @@ After=cockpit-session.socket
 ExecStart=@libexecdir@/cockpit-ws --no-tls --port=0
 DynamicUser=true
 Group=cockpit-session-socket
+
+PrivateDevices=true
+ProtectKernelTunables=true
+ProtectControlGroups=true
+MemoryDenyWriteExecute=true
+SystemCallFilter=@system-service
+
+# cockpit-tls does all our outside networking
+PrivateNetwork=yes
+PrivateIPC=yes
+RestrictAddressFamilies=AF_UNIX
+
+# extra protection for our TLS keys -- only cockpit-tls should read them
+InaccessiblePaths=-/etc/cockpit/ws-certs.d
```

Comment:
I don't care for this at all.  We're a dynamic user, so the only way we can read those is if they were world-readable.  But then we could read everyone else's incorrectly world-readable data.

I'd prefer if you take a much more aggressive approach here and move to whitelisting: we should only be allowed to see:
  - `/usr`
  - `/etc/cockpit/cockpit.conf`
  - the `/run/cockpit/session` socket

Reply:
Yup, I just pushed an initial attempt after some light testing. There are already some failures, so some work to do.

Locking this down further is nice of course. Need to see how that plays out with `PrivateTmp=`, i.e. which one wins.

Reply:
.. plus, need to allow whatever ssh needs to read, i. e. at least /etc/ssh/. It likely also requires reading $HOME for ssh keys..

Reply:
bastion host configurations have no business reading anyone's ssh keys from `$HOME`...

Reply:
I started with `InaccessiblePaths=/` but this is too much of a ratsnest. (Plus, it's not even documented to work, as these options don't have an ordering/priority). Think /etc/nsswitch.conf an ld.so.conf and whatnot. `ProtectSystem=strict` is a very good starting point, and together with `ProtectHome` (which is implied by DynamicUser) is a robust basis. You could spend endless hours tracking it down to a list of 50 files, but this will also change over time, is too specific to be reliable across different system configurations, and just isn't worth it IMHO -- having pretty much everything being read-only gives us what we need.

Reply:
Think about *some people* (cough, cough) using a Yubikey for SSH auth and requiring extra configs/libraries.

Reply:
> Think about some people (cough, cough) using a Yubikey for SSH auth and requiring extra configs/libraries.

This wouldn't require access to the Yubikey on the bastion host and, indeed, we'd actually want to very strictly isolate SSH on the bastion host for exactly this reason.

But there are some plausible causes where ssh on the bastion host would need to see more of the system:
 - we might imagine ssh ends up using systemd-resolved and needing to talk to it via D-Bus.  That's a whole lot of stuff that ends up getting pulled in.
 - we might imagine some sssd-type setup where we feed hostkeys to ssh in some dynamic way backed by some kind of a network database.  This could end up using a whole lot of random stuff around.

So, rough mid/longterm plan:
 - get cockpit-beiboot separated from -ws via the same socket activation tricks
 - after `ssh` is out of the picture, we can go insane on restrictions on -ws